### PR TITLE
More Samples in FIO

### PR DESCRIPTION
**Pull request by noahl**
Make the FIO benchmark add samples for each latency statistic, in
addition to putting all of the statistics in the metadata of a single
sample as before.

### CODE REVIEW

**Comment by noahl** [comment]
@cmccoy @yuyantingzero Can you take a look?

File: perfkitbenchmarker/packages/fio.py [hunk 1]
```diff
@@ -170,6 +177,49 @@ def ParseResults(job_file, fio_json_result):
             sample.Sample('%s:latency' % metric_name,
                           job[mode]['clat']['mean'],
                           'usec', lat_metadata))
+
+        samples.append(
```
Comment (yuyantingzero) [inline perfkitbenchmarker/packages/fio.py]: Why tream min, max, mean stddev seperately?

File: perfkitbenchmarker/packages/fio.py [hunk 2]
```diff
@@ -170,6 +177,49 @@ def ParseResults(job_file, fio_json_result):
...
+            sample.Sample('%s:latency:mean' % metric_name,
+                          job[mode]['clat']['mean'],
+                          'usec', parameters))
+        samples.append(
+            sample.Sample('%s:latency:stddev' % metric_name,
+                          job[mode]['clat']['stddev'],
+                          'usec', parameters))
+        # these tuples map the name we will record in our sample to
+        # the name fio gave us in the JSON results
+        latency_statistic_name_pairs = [
```
Comment (yuyantingzero) [inline perfkitbenchmarker/packages/fio.py]: This probably can be a constant so we can also refactor line 154-174

File: perfkitbenchmarker/packages/fio.py (hunk 1 above)
Comment (noahl) [inline perfkitbenchmarker/packages/fio.py]: Changed in the new version.

File: perfkitbenchmarker/packages/fio.py (hunk 2 above)
Comment (noahl) [inline perfkitbenchmarker/packages/fio.py]: Done.

File: perfkitbenchmarker/packages/fio.py [hunk 3]
```diff
@@ -144,32 +144,49 @@ def ParseResults(job_file, fio_json_result):
...
-            'p99': job[mode]['clat']['percentile']['99.000000'],
-            'p99.5': job[mode]['clat']['percentile']['99.500000'],
-            'p99.9': job[mode]['clat']['percentile']['99.900000'],
-            'p99.95': job[mode]['clat']['percentile']['99.950000'],
-            'p99.99': job[mode]['clat']['percentile']['99.990000']}
-        lat_metadata.update(parameters)
+
+        # There is one sample whose metric is '<metric_name>:latency'
+        # with all of the latency statistics in its metadata, and then
+        # ten samples whose metrics are '<metric_name>:latency:min'
```
Comment (cmccoy) [inline perfkitbenchmarker/packages/fio.py]: This looks like more than ten samples.

File: perfkitbenchmarker/packages/fio.py [hunk 4]
```diff
@@ -144,32 +144,49 @@ def ParseResults(job_file, fio_json_result):
...
-            'p90': job[mode]['clat']['percentile']['90.000000'],
-            'p95': job[mode]['clat']['percentile']['95.000000'],
-            'p99': job[mode]['clat']['percentile']['99.000000'],
-            'p99.5': job[mode]['clat']['percentile']['99.500000'],
-            'p99.9': job[mode]['clat']['percentile']['99.900000'],
-            'p99.95': job[mode]['clat']['percentile']['99.950000'],
-            'p99.99': job[mode]['clat']['percentile']['99.990000']}
-        lat_metadata.update(parameters)
+
+        # There is one sample whose metric is '<metric_name>:latency'
```
Comment (cmccoy) [inline perfkitbenchmarker/packages/fio.py]: Nit: could you introduce locals for `job[mode]['clat']` and `job[mode]['clat']['percentile']`?

File: perfkitbenchmarker/packages/fio.py (hunk 3 above)
Comment (noahl) [inline perfkitbenchmarker/packages/fio.py]: Comment fixed

File: perfkitbenchmarker/packages/fio.py (hunk 4 above)
Comment (noahl) [inline perfkitbenchmarker/packages/fio.py]: Done

**Comment by cmccoy** [comment]
LGTM.

**Pull request merged**